File: app/serializers/search_result_serializer.rb
class SearchResultSerializer < BaseSerializer
  attributes :title, :url, :tag, :search_tokens, :type

  delegate :url, :search_tokens, :type, to: :object

  def title
    object.short_title
  end

  def tag
    object.is_snippet? ? object.formatted_mini_preview_tag : object.formatted_snippet_count
  end
end

File: src/serializers/searchResultSerializer.js
import Serializer from '#src/core/serializer.js';

export default class SearchResultSerializer extends Serializer {
  static {
    Serializer.prepare(this, [
      ['title', 'shortTitle'],
      'url',
      [
        'tag',
        object => {
          return object.isSnippet
            ? object.formattedMiniPreviewTag
            : object.formattedSnippetCount;
        },
      ],
      ['searchTokens', 'docTokens'],
      'type',
    ]);
  }
}
